token-swap/program/src/curve/stable.rs

```diff
@@ -35,21 +40,43 @@ pub struct StableCurve {
     pub amp: u64,
 }
 
+construct_uint! {
+    pub struct U256(4);
+}
+
+fn checked_u8_power(a: &U256, b: u8) -> Option<U256> {
+    let mut result = *a;
+    for _ in 1..b {
+        result = result.checked_mul(*a)?;
+    }
+    Some(result)
+}
+
+fn checked_u8_mul(a: &U256, b: u8) -> Option<U256> {
+    let mut result = *a;
+    for _ in 1..b {
+        result = result.checked_add(*a)?;
+    }
+    Some(result)
+}
+
+fn almost_equal(a: &U256, b: &U256) -> bool {
+    if a > b {
+        a - b <= U256::one()
+    } else {
+        b - a <= U256::one()
```

Comment:
nit: I know that you do the check earlier to make sure that isn't a problem, but for security reviews, could you do checked math here?